sdk/storage/azure-storage-files-shares/src/share_client.cpp

```diff
@@ -41,31 +41,44 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       const ShareClientOptions& options)
       : m_shareUrl(shareUrl)
   {
-    Azure::Core::Http::TelemetryOptions telemetryPolicyOptions;
-    telemetryPolicyOptions.ApplicationId = options.ApplicationId;
+    ShareClientOptions newOptions = options;
+    newOptions.PerRetryPolicies.insert(
+        newOptions.PerRetryPolicies.begin(),
+        std::make_unique<Storage::Details::StoragePerRetryPolicy>());
+    newOptions.PerRetryPolicies.emplace_back(
+        std::make_unique<Storage::Details::SharedKeyPolicy>(credential));
+    {
+      Azure::Core::Http::Internal::ValueOptions valueOptions;
+      valueOptions.HeaderValues[Storage::Details::HttpHeaderXMsVersion] = newOptions.ApiVersion;
+      newOptions.PerOperationPolicies.insert(
+          newOptions.PerOperationPolicies.begin(),
+          std::make_unique<Azure::Core::Http::Internal::ValuePolicy>(valueOptions));
+    }
```

Comment:
Don't insert the service policies into the end-user policies.  Pass the service-policies as a parameter to the HttpPipeline constructor

Reply:
okay, although I don't think it's necessary to separate them.